engine/src/main/java/org/hibernate/search/engine/common/impl/SearchIntegrationPartialBuildStateImpl.java

```diff
@@ -164,11 +172,30 @@ public SearchIntegration finalizeIntegration() {
 			}
 			failureCollector.checkNoFailure();
 
+			// Pre-Start indexes
+			for ( Map.Entry<String, IndexManagerNonStartedState> entry : nonStartedIndexManagers.entrySet() ) {
+				SavedState savedState = null;
+				if ( previousIntegration.isPresent() ) {
+					Map<String, SavedState> indexManagersStates = previousIntegration.get().saveForRestart()
+							.get( INDEX_MANAGERS_KEY ).orElse( Collections.emptyMap() );
+
+					savedState = indexManagersStates.get( entry.getKey() );
+				}
+
+				entry.getValue().preStart( failureCollector, beanResolver, propertySource,
+						( savedState == null ) ? SavedState.empty() : savedState );
```

Comment:
Running the tests, it seems that the null check is still required, since with change the default is applied only if `previousIntegration.isPresent()`. I'm reverting it...

Reply:
Or you could replace `SavedState savedState = null;` with `SavedState savedState = SavedState.empty();`.

Reply:
Right, we can init the property using the empty value.